R/plotNhoods.R

```diff
@@ -191,6 +191,101 @@ plotNhoodGraphDA <- function(x, milo_res, alpha=0.05, ... ){
   plotNhoodGraph(x, colour_by = "logFC", ... )
   }
 
+#' Visualize gene expression in neighbourhoods
+#'
+#' Plots the average gene expression in neighbourhoods, sorted by DA fold-change
+#'
+#' @param x A \code{\linkS4class{Milo}} object
+#' @param da.res a data.frame of DA testing results
+#' @param features a character vector of features to plot (they must be in rownames(x))
+#' @param alpha significance level for Spatial FDR (default: 0.1)
+#' @param subset.nhoods indices of subset of nhoods to show in plot (default: NULL, no subsetting)
+#' @param cluster_features logical indicating whether features should be clustered with hierarchical clustering.
+#' If FALSE then the order in \code{features} is maintained (default: FALSE)
+#' @param assay A character scalar that describes the assay slot to use for calculating neighbourhood expression.
+#' (default: logcounts)
+#'
+#' @return a \code{\linkS4class{ggplot}} object
+#'
+#' @author Emma Dann
+#'
+#' @examples
+#' NULL
+#'
+#' @export
+#' @rdname plotNhoodExpressionDA
+#' @import ggplot2
+#' @import patchwork
+#' @import dplyr
+plotNhoodExpressionDA <- function(x, da.res, features, alpha=0.1,
+                                  subset.nhoods=NULL, cluster_features=FALSE, assay="logcounts"){
+  ## Check if features are in rownames(x)
+  if (!all(features %in% rownames(x))) {
+    stop("Some features are not in rownames(x)")
+  }
+  ## Check if nhood expression exists 
+  if (dim(nhoodExpression(x))[2] == 1){
+    warning("Nothing in nhoodExpression(x): computing for requested features...")
+    x <- calcNhoodExpression(x, assay = assay, subset.row = features)
+  } 
+  ## Check if all features are in nhoodExpression
+  if (!all(features %in% rownames(nhoodExpression(x)))) {
+    warning("Not all features in nhoodExpression(x): recomputing for requested features...")  
+    x <- calcNhoodExpression(x, assay = assay, subset.row = features)
+  } 
+  ## Get nhood expression matrix
+  if (is.null(subset.nhoods)) {
+    expr_mat <- nhoodExpression(x)[features,]
+    colnames(expr_mat) <- 1:length(nhoods(x))
+  } else {
+    expr_mat <- nhoodExpression(x)[features,subset.nhoods]
+    colnames(expr_mat) <- subset.nhoods
```

Comment:
Can you extend this to allow `subset.nhoods` to be either a logical, character or numeric so that it matches the equivalent behaviour of `subset.rows` more generally?

Reply:
In which case would a logical be needed?

Reply:
If you pass a vector of say, HVGs but where you pass a logical vector of which genes are HVGs instead of their indices or gene names.